Wikipedia/Code/WMFArticleImageGalleryDataSource.m

```diff
@@ -26,7 +27,29 @@ @implementation WMFArticleImageGalleryDataSource
 @dynamic emptyView;
 
 - (instancetype)initWithArticle:(MWKArticle*)article {
-    self = [super wmf_initWithItemsAndReverseIfNeeded:article.images.uniqueLargestVariants];
+    
+    NSArray* images = [article.images.uniqueLargestVariants bk_select:^BOOL (MWKImage* image) {
+        // Keep image if it's the article image even if we can't determine its size - we always want to show article image as first "lead" image by gallery.
+        if ([article.image isEqualToImage:image]) {
+            return YES;
+        }
+        
+        if (!image.width || !image.height) {
+            // HAX: if this image MWKImage record doesn't have a width and height value (because it wasn't determined by parsing it
+            // from the article HTML's image url) see if the cache can tell us the size.
+            SDImageCache* myCache = [SDImageCache wmf_appSupportCacheWithNamespace:@"default"];
+            UIImage* img = [myCache imageFromDiskCacheForKey:image.sourceURLString];
```

Comment:
We should expose this method on WMFImageController so we don't need to work with the SDImageCache directly.